projects/week_1/Instructions.md

```diff
@@ -0,0 +1,332 @@
+
+### 2.7 Transform tag position
+
+We now have two transformations we need to apply to our tags. The first we grabbed for you from the ROS TF system (which we'll cover in detail later). This is `camera_to_base_transform`, which transforms things from the camera's conventional frame to the robot's base frame. The second transform, `camera_optical_to_conventional_transform` is the rotation transformation you created in the `getRotationMatrixForOpticalFrame()` function. Our objective is to transform all of the tag detections from the camera's optical frame to the robot's base frame, so we need to apply both of these transformations to the position and orientation of each tag. In this section, we'll handle the position.
+
+Let's start by creating a homogeneous vector for our tag's position. In the body of our for loop, after the `new_tag.id = tag_array_msg->tags[i].id;` line, declare a variable of type `Eigen::Vector4d`, named `position`. Initialize `position` by passing, to its constructor, the x, y, and z coordinates from the old tag's pose, and 1 for the final element. That will look like this:
+
+```c++
+Eigen::Vector4d position(
+  tag_array_msg->tags[i].pose.position.x,
+  tag_array_msg->tags[i].pose.position.y,
+  tag_array_msg->tags[i].pose.position.z,
+  1
+);
+```
+
+Now, apply the transformations by multiplying the transformation objects, `camera_to_base_transform` and `camera_optical_to_conventional_transform`, with `position`. Assign the result of that multiplication back to `position`. Remember, the order of this multiplication matters. Be sure to apply `camera_optical_to_conventional_transform` before `camera_to_base_transform`.
+
+The last thing we need to do for this section is to copy the new position's coordinates into the `new_tag` message. Copying the x value looks like this:
+
+```c++
+new_tag.pose.position.x = position.x();
+```
+
+Do the same for y and z.
+
+### 2.8 Transform tag orientation
+
+To transform the tag's orientation, we're going to follow a similar process. We'll start by extracting the rotation from the old tag message. We've provided a function to help you with this. Declare a variable named `tag_orientation` of type `Eigen::Matrix4d` right after you position code. Initialize `tag_orientation` by calling `quaternionMessageToTransformationMatrix()`, passing it `tag_array_msg->tags[i].pose.orientation`.
+
+Now, apply both transformations to `tag_orientation`. This should look basically identical to how you applied the transformations to `position`.
+
+Finally, we need to convert our new orientation back into a quaternion message and set it in the `new_tag` message. Here again, we've got a helper function for you called `transformationMatrixToQuaternionMessage()`. Call this function, passing `tag_orientation`, and assign the result to `new_tag.pose.orientation`.
+
+And that's it! We've written all the code we need to make this node complete.
```

Comment:
This section feel a little weird to me since you are using a 4x4 when you only need a 3x3. I get the code is simpler but I don't want to confuse them. We can talk about this tonight. 

Maybe something as simple as explaining the above point would be good enough for me. i.e. that the translation vector in the 4x4 block matrix doesn't matter here.

Reply:
I think something simple like saying the rotation is represented by the rotation matrix in the transformation matrix.